hadoop-pipeline-application/src/main/java/org/molgenis/hadoop/pipeline/application/DistributedCacheHandler.java

```diff
@@ -88,33 +92,34 @@ public void addCacheToJob(CommandLineInputParser parser)
 	}
 
 	/**
-	 * {@link URI} of the tools archive stored in {@link JobContext#getCacheArchives()}.
+	 * {@link String} of the tools archive stored in {@link JobContext#getCacheArchives()}.
 	 * 
 	 * @return {@link String}
 	 * @throws IOException
 	 */
 	public String getToolsArchive() throws IOException
 	{
-		return FilenameUtils.getName(context.getCacheArchives()[0].toString());
+		return getArchiveFromCache(0);
```

Comment:
You can use static instance variables to clearly represent these numbers:
```java
private static final int TOOL_ARCHIVE = 0;
```

Same for the files:
```java
private static final int TOOL_ARCHIVE = 0;
private static final int FASTA_AMB = 1;
etc.
```